src/main/scala/it/pagopa/interop/agreementprocess/api/impl/AgreementApiServiceImpl.scala

```diff
@@ -625,21 +630,28 @@ final case class AgreementApiServiceImpl(
             .map(_ => ())
         else Future.unit
 
-      // * Se lo stato non è attivo a causa dei flags ritorna un errore applicativo
-      // * Per questo viene fatto il controllo a valle
+      // * If the state is not active due to flags this will return an applicative error
+      // * That's why this check is performed downstream
       _ <- validateResultState(newState)
-      _ <- authorizationManagementService.updateStateOnClients(
-        eServiceId = agreement.eserviceId,
-        consumerId = agreement.consumerId,
-        agreementId = agreement.id,
-        state = toClientState(newState)
-      )
+      _ <-
+        if (activeOrSuspended(newState))
+          authorizationManagementService.updateStateOnClients(
+            eServiceId = agreement.eserviceId,
+            consumerId = agreement.consumerId,
+            agreementId = agreement.id,
+            state = toClientState(newState)
+          )
+        else Future.unit
       _ <-
         if (updated.state == AgreementManagement.AgreementState.ACTIVE && agreements.isEmpty)
           createContractAndSendMail(updated, updateSeed)
         else Future.unit
     } yield updated
+  }
 
+  def activeOrSuspended(newState: AgreementManagement.AgreementState): Boolean = newState match {
+    case AgreementManagement.AgreementState.ACTIVE | AgreementManagement.AgreementState.SUSPENDED => true
+    case _                                                                                        => false
   }
```

Comment:
just minor consideration
```suggestion
  def activeOrSuspended(newState: AgreementManagement.AgreementState): Boolean =
    newState == AgreementManagement.AgreementState.ACTIVE || newState == AgreementManagement.AgreementState.SUSPENDED 
```